core/src/main/java/org/infinispan/protostream/TagHandler.java

```diff
@@ -13,8 +15,10 @@
 
    /**
     * Start of top level message. Do whatever required init here.
+    *
+    * @param descriptor The {@link GenericDescriptor} for the top level Enum or Message. For primitives it is null.
     */
-   default void onStart() {
+   default void onStart(AnnotatedDescriptor descriptor) {
```

Comment:
I gave some more thought about AnnotationDescriptor vs GenericDescriptor in onStart method. Going with AnnotationDescriptor seems a bit easier but is not very orthogonal and also does not adhere to the initial javadoc of TagHandler.onStart  :)  So let's go with GenericDescriptor :)

I've done this change and pushed a branch here https://github.com/anistor/protostream/commits/t_gustavonalle_json   (see last commit).